examples/graphics/mesh-decals.html

```diff
@@ -0,0 +1,245 @@
+        });
+
+        app.scene.ambientLight = new pc.Color(0.2, 0.2, 0.2);
+
+        // create plane primitive
+        var primitive = new pc.Entity();
+        primitive.addComponent('model', {
+            type: "plane"
+        });
+
+        // set position and scale and add it to scene
+        primitive.setLocalScale(new pc.Vec3(20, 20, 20));
+        primitive.setLocalPosition(new pc.Vec3(0, -0.01, 0));
+        app.root.addChild(primitive);
+
+        // Create an Entity with a point light component 
+        var light = new pc.Entity();
+        light.addComponent("light", {
+            type: "point",
+            color: new pc.Color(0.2, 0.2, 0.2),
+            range: 30,
+            castShadows : true
+        });
+        light.translate(0, 8, 0);
+        app.root.addChild(light);
+
+        // Create an Entity with a camera component
+        var camera = new pc.Entity();
+        camera.addComponent("camera", {
+            clearColor: new pc.Color(0.2, 0.2, 0.2)
+        });
+
+        // Add the camera to the hierarchy
+        app.root.addChild(camera);
+
+        // Position the camera
+        camera.translate(0, 10, 20);
+        camera.lookAt(pc.Vec3.ZERO);
+
+        // Create bounding call model and add it to hierarchy
+        this.ball = new pc.Entity();
+        this.ball.addComponent("model", {
+            type: "sphere",
+        });
+        app.root.addChild(this.ball);
+
+        // Allocate space for decals. Each decal is a quad with 4 verticies
+        this.numDecals = 500;
+        var numDecalVertices = 4 * this.numDecals;
+
+        // Allocate storage for vertex positions, vertex stores x, y and z
+        this.positions = new Float32Array(3 * numDecalVertices);
+
+        // Allocate storage for colors, each vertex stores r, g, b and a
+        this.colors = new Uint8ClampedArray(4 * numDecalVertices);
+
+        // Allocate storage for uvs, each vertex stores u and v. And fill them up to display whole texture
+        this.uvs = [];
+        var i;
+        for (i = 0; i < this.numDecals; i++)
+            uvs.push(0, 0,      0, 1,       1, 1,       1, 0);
+
+        // Allocate and generate indices. Each quad is representing using 2 triangles, and uses 4 vertices
+        var quadTriangles = [0, 1, 2,       2, 3, 0];
+        this.indices = new Uint16Array(6 * this.numDecals);
+        for (i = 0; i < this.numDecals; i++) {
+            this.indices[6 * i + 0] = 4 * i + quadTriangles[0];     
+            this.indices[6 * i + 1] = 4 * i + quadTriangles[1];
+            this.indices[6 * i + 2] = 4 * i + quadTriangles[2];
+            this.indices[6 * i + 3] = 4 * i + quadTriangles[3];
+            this.indices[6 * i + 4] = 4 * i + quadTriangles[4];
+            this.indices[6 * i + 5] = 4 * i + quadTriangles[5];
+        }
+
+        // Helper function to generate a decal with index i at position pos. It fills up information for all 4 vertices of a quad
+        function CreateDecal(i, pos) {
```

Comment:
CreateDecal -> createDecal (stick to camel case)

Reply:
fixing, thanks